display_list/display_list_builder.cc

```diff
@@ -636,11 +635,10 @@ void DisplayListBuilder::transform2DAffine(
     Push<Transform2DAffineOp>(0, 1,
                               mxx, mxy, mxt,
                               myx, myy, myt);
-    current_layer_->matrix().preConcat(SkM44(mxx, mxy,  0,  mxt,
-                                             myx, myy,  0,  myt,
-                                             0,   0,   1,   0,
-                                             0,   0,   0,   1));
-    current_layer_->update_matrix33();
+    tracker_.transform(SkM44(mxx, mxy,  0,  mxt,
```

Comment:
Use `transform(SkMatrix)` otherwise you promote it to an SkM44 before it needs it.

If you want to avoid scalars->SkMatrix->SkM44, then you could ask it if it is already an M44 and call either `transform(SkMatrix)` or `transform(SkM44)` accordingly.

And maybe we should have transform(6 scalars), transform(16 scalars) in the Tracker so that it can figure out which to use without the caller having to ask.

Reply:
Done.